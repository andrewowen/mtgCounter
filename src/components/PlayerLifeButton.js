import React from 'react';
import { StyleSheet, Text, TouchableOpacity } from 'react-native';

const PlayerLifeButton = props => {
  return (
    <TouchableOpacity
      style={styles.playerLifeButton}
      onPress={props.onPress}
      disabled={props.currentLife === 0}
    >
      <Text style={{ fontSize: 18, color: '#fff' }}>{props.buttonText}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  playerLifeButton: {
    flex: 1,
    color: '#fff',
    borderRadius: 10,
    width: '100%',
    justifyContent: 'center',
    alignItems: 'center'
  }
});

export default PlayerLifeButton;
