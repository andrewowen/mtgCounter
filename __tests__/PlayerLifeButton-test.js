import React from 'react';
import PlayerLifeButton from '../src/components/PlayerLifeButton';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<PlayerLifeButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
