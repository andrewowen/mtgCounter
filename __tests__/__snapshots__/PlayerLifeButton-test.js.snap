// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 10,
      "color": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 18,
      }
    }
  />
</View>
`;
